pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1234,23 +1257,23 @@ func storePipelineSpecAndMergeMeta(ctx context.Context, pr *v1beta1.PipelineRun,
 	cfg := config.FromContextOrDefaults(ctx)
 	if cfg.FeatureFlags.EnableProvenanceInStatus {
 		if pr.Status.Provenance == nil {
-			pr.Status.Provenance = &v1beta1.Provenance{}
+			pr.Status.Provenance = &v1.Provenance{}
 		}
 		// Store FeatureFlags in the Provenance.
 		pr.Status.Provenance.FeatureFlags = cfg.FeatureFlags
 
 		if meta != nil && meta.RefSource != nil && pr.Status.Provenance.RefSource == nil {
 			pr.Status.Provenance.RefSource = meta.RefSource
 		}
-		if meta != nil && meta.RefSource != nil && pr.Status.Provenance.ConfigSource == nil {
-			pr.Status.Provenance.ConfigSource = (*v1beta1.ConfigSource)(meta.RefSource)
+		if meta != nil && meta.RefSource != nil && pr.Status.Provenance.RefSource == nil {
```

Comment:
it looks like this is repeating the last if statement

Reply:
Thanks, didn't catch this after replacing the deprecated ConfigSource to RefSource.